### PR TITLE
EN-30766 Return truth data version in UpserResult

### PR DESCRIPTION
**Pull request by rozap**
* which will be passed back through core

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -11,7 +11,7 @@ organization := "com.socrata"
 name := "soda-api-java"
 
 // remember to update the README when you change this to a new release
```
Comment (branweb1) [inline build.sbt]: :point_up: ?

File: src/main/java/com/socrata/api/Soda2Producer.java [hunk 2]
```diff
@@ -520,14 +520,20 @@ UpsertResult deserializeUpsertResult(InputStream is) throws IOException
...
         }
 
         return parser.readValueAs(UpsertResult.class);
     }
 
     UpsertResult deserializeUpsertResult(Response response) throws IOException {
-        return deserializeUpsertResult(response.readEntity(InputStream.class));
+        Long truthDataVersion = null;
+        try {
+            truthDataVersion = Long.parseLong(response.getHeaders().getFirst("X-SODA2-Truth-Version").toString());
```
Comment (rjmac) [inline src/main/java/com/socrata/api/Soda2Producer.java]: This header might not exist (e.g., if the upsert is being done on an OBE dataset) in which case you'll get a nullpointerexception here

File: src/main/java/com/socrata/api/Soda2Producer.java [hunk 3]
```diff
@@ -520,14 +520,20 @@ UpsertResult deserializeUpsertResult(InputStream is) throws IOException
...
+        } catch (ClassCastException cce) {
```
Comment (rjmac) [inline src/main/java/com/socrata/api/Soda2Producer.java]: ..you should never get a ClassCastException there, though.

Comment (rjmac) [reply src/main/java/com/socrata/api/Soda2Producer.java]: ..and in theory you _could_ get a NumberFormatException

**Pull request merged**